scripts/zones/Abyssea-Konschtat/mobs/Hadal_Satiator.lua

```diff
@@ -0,0 +1,27 @@
+-----------------------------------
+-- Area: Abyssea-Konschtat
+--  Mob: Hadal Satiator
+-----------------------------------
+mixins = { require('scripts/mixins/families/gorger_nm') }
+-----------------------------------
+---@type TMobEntity
+local entity = {}
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    local tpMoves =
+    {
+        xi.mobSkill.FISSION,
+        xi.mobSkill.PROMYVION_BARRIER_2,
+        xi.mobSkill.QUADRATIC_CONTINUUM_2,
+        xi.mobSkill.SPIRIT_ABSORPTION_2,
+        xi.mobSkill.STYGIAN_FLATUS_1,
+        xi.mobSkill.VANITY_DRIVE_2,
+    }
+
+    -- Filter out Fission if all adds are already spawned
+    tpMoves = xi.mix.gorger.filterFission(mob, tpMoves)
+
+    return tpMoves[math.random(#tpMoves)]
+end
```

Comment:
Here. Remove Fission from the table. Instead check if fission is an option.
After checking if fission is an option, if it is, insert it to the table directly and then perform the random.

I think this method should simplify the logic a lot (in the global) and opens up the possibility to keep editing the move list and weight the table further down the line